rasa/core/training/story_reader/yaml_story_reader.py

```diff
@@ -0,0 +1,245 @@
+import logging
+from typing import Dict, Text, List, Any
+
+import rasa.utils.io
+from rasa.constants import DOCS_URL_STORIES
+from rasa.core.constants import INTENT_MESSAGE_PREFIX
+from rasa.core.events import UserUttered, SlotSet
+from rasa.core.training.story_reader.story_reader import StoryReader
+from rasa.core.training.structures import StoryStep
+from rasa.data import YAML_FILE_EXTENSIONS
+from rasa.utils.common import raise_warning
+
+logger = logging.getLogger(__name__)
+
+KEY_STORIES = "stories"
+KEY_PLAIN_STORY_NAME = "story"
+KEY_PLAIN_STORY_STEPS = "steps"
+KEY_PLAIN_STORY_USER_UTTERANCE = "user"
+KEY_SLOT_NAME = "slot"
+KEY_SLOT_VALUE = "value"
+KEY_ACTION = "action"
+KEY_PLAIN_STORY_ENTITIES = "entities"
+KEY_CHECKPOINT = "checkpoint"
+KEY_CHECKPOINT_SLOTS = "slots"
+KEY_METADATA = "metadata"
+KEY_OR = "or"
+
+
+class YAMLStoryReader(StoryReader):
+    """Class that reads the core training data in a YAML format
+
+    """
+
+    async def read_from_file(self, filename: Text) -> List[StoryStep]:
+        try:
+            yaml_content = rasa.utils.io.read_yaml_file(filename)
+
+            stories = yaml_content.get(KEY_STORIES)  # pytype: disable=attribute-error
+            if not stories:
+                return []
+
+            self._parse_stories(stories)
+            self._add_current_stories_to_result()
+            return self.story_steps
+
+        except ValueError:
+            logger.error(f"Failed to read {filename}, it will be skipped")
+
+        return []
+
+    def _parse_stories(self, stories: List[Dict[Text, Any]]) -> None:
+        for story_item in stories:
+            if not isinstance(story_item, dict):
+                raise_warning(
```

Comment:
let's not import the function directly but use the module to be consistent with our python conventions

Reply:
I'm not sure how the `common_utils.raise_warning()` improve the readability here, in the absence of other `raise_warning`, but sure.

Reply:
Well, it's harder to see that this function was imported and is not part of the module. But I see your point, maybe we can discuss in https://www.notion.so/Python-Code-Conventions-26835ffd72484dd699f28fa693916d2e where we always do that or whether we make an exception of some utility functions